source/common/runtime/runtime_impl.cc

```diff
@@ -23,60 +23,78 @@ namespace Runtime {
 
 const size_t RandomGeneratorImpl::UUID_LENGTH = 36;
 
+void RandomGeneratorImpl::random_bytes(uint8_t* out, size_t out_len) {
+  if (out_len >= 1024) {
+    int rc = RAND_bytes(out, out_len);
+    ASSERT(rc == 1);
+    UNREFERENCED_PARAMETER(rc);
+    return;
+  }
+
+  static thread_local uint8_t buffered[2048];
+  static thread_local size_t buffered_idx = sizeof(buffered);
```

Comment:
It took me a little bit of time to figure out what was going on here. Can you add small comment that says initialized to sizeof(buffered) will cause reload on first iteration.